src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/factory/eventGetContext.hpp

```diff
@@ -0,0 +1,138 @@
+ *
+ * This handler queries the indexer for system inventory context information.
+ * It executes a GET query to retrieve context data from inventory indices.
+ *
+ * @tparam TIndexerConnector indexer connector type.
+ * @tparam TScanContext scan context type.
+ */
+template<typename TIndexerConnector = IndexerConnectorSync, typename TScanContext = ScanContext>
+class TEventGetContext final : public AbstractHandler<std::shared_ptr<TScanContext>>
+{
+private:
+    std::shared_ptr<TIndexerConnector> m_indexerConnector;
+    static constexpr size_t DEFAULT_PAGE_SIZE = 1000;
+public:
+    /**
+     * @brief EventGetContext constructor.
+     *
+     * @param indexerConnector Indexer connector.
+     */
+    explicit TEventGetContext(std::shared_ptr<TIndexerConnector> indexerConnector)
+        : m_indexerConnector(std::move(indexerConnector))
+    {
+    }
+
+    /**
+     * @brief Handles request and passes control to the next step of the chain.
+     *
+     * Executes a GET query to the indexer to retrieve system inventory context data
+     * across multiple indices:
+     * - wazuh-states-inventory-system
+     * - wazuh-states-inventory-packages
+     * - wazuh-states-inventory-hotfixes
+     * - wazuh-states-vulnerabilities
+     *
+     * @param data Scan context.
+     * @return std::shared_ptr<TScanContext> Scan context passed to next handler.
+     */
+    // Helper to safely extract _id from a hit
+    static const std::string* getContextIdPtr(const nlohmann::json& hit) {
+        if (const auto* idPtr = hit.find("_id"); idPtr != hit.end() && idPtr->is_string()) {
+            return &idPtr->get_ref<const std::string&>();
+        }
+        return nullptr;
+    }
+
+    std::shared_ptr<TScanContext> handleRequest(std::shared_ptr<TScanContext> data) override
+    {
+        logDebug2(WM_VULNSCAN_LOGTAG, "TEventGetContext - Starting GET context query handler");
+
+        try
+        {
+            // Define the indices to query
+            const std::vector<std::string> indexes = {
+                "wazuh-states-inventory-system",
+                "wazuh-states-inventory-packages",
+                "wazuh-states-inventory-hotfixes",
+                "wazuh-states-vulnerabilities"
+            };
+
+            int totalContextItems = 0;
+
+            logDebug2(WM_VULNSCAN_LOGTAG, "TEventGetContext - Building context query");
+
+            // Build query without search_after - pagination is now automatic
+            auto query = InventorySyncQueryBuilder::buildContextGetQuery(
+                std::string(data->agentId()), 
+                DEFAULT_PAGE_SIZE,
+                ""); // Empty search_after - first page
+
+            // Execute query for each index separately
+            for (const auto& index : indexes)
+            {
+                m_indexerConnector->executeSearchQueryWithPagination(index, query,
+                    [totalContextItems, data](const nlohmann::json& response) mutable {
```

Comment:
Isn't the `totalContextItems` variable a copy? Wouldn't it be necessary to capture it by reference?